orttraining/orttraining/python/training/ortmodule.py

```diff
@@ -236,14 +250,14 @@ def backward(ctx, *grad_outputs):
                     backward_outputs = training_io_binding.get_outputs()
 
                     # Return input and initializer gradients
-                    num_user_input_grads = len(self._input_names_require_grad)
+                    num_user_input_grads = len(self._input_info.require_grad_names)
 
                     results = []
                     for input_name in self._onnx_graphs_info.user_input_names:
                         try:
                             # Append to the results the backward output for each input that required grad
                             results.append(_ortvalue_to_torch_tensor(
-                                backward_outputs[self._input_names_require_grad.index(input_name)]))
+                                backward_outputs[self._input_info.require_grad_names.index(input_name)]))
```

Comment:
As part of the refactoring in this PR, can we replace (or supplement) the `require_grad_names` list with a set, so that this loop isn't O(n^2) in the number of tensors that require a gradient?

Reply:
Sure, this change came from the original YieldOp merge, but I can take a look into it